### PR TITLE
split bulk payloads to avoid max body limit

### PR DESCRIPTION
**Pull request by snopoke**
@czue @esoergel 

cc @NoahCarnahan

### CODE REVIEW

**Comment by NoahCarnahan** [comment]
:+1:

**Comment by czue** [comment]
:+1: makes sense

**Comment by czue** [comment]
actually meant to put that comment on https://github.com/dimagi/commcare-hq/pull/9966 but this makes sense too. also nice to see tests.

File: corehq/ex-submodules/pillowtop/utils.py [hunk 1]
```diff
@@ -135,3 +140,18 @@ def get_pillow_json(pillow_config):
...
+    payloads = ['']
+    for bulk_chunk in chunked(bulk_changes, chunk_size):
+        current_payload = payloads[-1]
+        payload_chunk = '\n'.join(map(simplejson.dumps, bulk_chunk)) + '\n'
+        appended_payload = current_payload + payload_chunk
+        new_payload_size = sys.getsizeof(appended_payload)
+        if new_payload_size > max_size:
+            payloads.append(payload_chunk)
+        else:
+            payloads[-1] = appended_payload
```
Comment (esoergel) [inline corehq/ex-submodules/pillowtop/utils.py]: :stars:

**Comment by esoergel** [comment]
Big fan of this change!
:guitar: :star2:

**Pull request merged**